### ISSUE TITLE
create接口 在cordova ios程序中报错

### ISSUE THREAD

**Issue by jackieron** [chanind/hanzi-writer#124]
我在cordova中使用这个库，当调用create接口是，会报错，loadchar错误，内核是使用的wkwebview 请问如何解决？谢谢

**Comment by jackieron** [comment]
the error is : Unhandled Promise Rejection: Error: Failed to load char data for 中  in hanzi-writer.min.js:1641
the hanziwriter version is v2.0.2

**Comment by chanind** [comment]
hrm it's hard to tell from that error why it's unable to load character data. Is networking restricted inside of cordova? If you try running the following js code inside of cordova, is it able to load character data?

```js
fetch('https://cdn.jsdelivr.net/npm/hanzi-writer-data@2.0.1/中.json').then(res => res.json()).then(console.log)
```

**Comment by jackieron** [comment]
Thank you very much for your reply! Can I download all Chinese character files locally and load them directly from local? Is there an interface for local loading? I want my app to work without the Internet.

**Comment by jackieron** [comment]
> hrm it's hard to tell from that error why it's unable to load character data. Is networking restricted inside of cordova? If you try running the following js code inside of cordova, is it able to load character data?
> 
> ```js
> fetch('https://cdn.jsdelivr.net/npm/hanzi-writer-data@2.0.1/中.json').then(res => res.json()).then(console.log)
> ```



> hrm it's hard to tell from that error why it's unable to load character data. Is networking restricted inside of cordova? If you try running the following js code inside of cordova, is it able to load character data?
> 
> ```js
> fetch('https://cdn.jsdelivr.net/npm/hanzi-writer-data@2.0.1/中.json').then(res => res.json()).then(console.log)
> ```
Thank you very much for your reply! Can I download all Chinese character files locally and load them directly from local? Is there an interface for local loading? I want my app to work without the Internet.

**Comment by chanind** [comment]
Yes, definitely! All the character data is in this repo: https://github.com/chanind/hanzi-writer-data. You can load data locally - you just need to provide your own `charDataLoader` function. More info here: https://chanind.github.io/hanzi-writer/docs#loading-character-data-link

**Comment by jackieron** [comment]
> Yes, definitely! All the character data is in this repo: https://github.com/chanind/hanzi-writer-data. You can load data locally - you just need to provide your own `charDataLoader` function. More info here: https://chanind.github.io/hanzi-writer/docs#loading-character-data-link

Thank you very much for your help. I have implemented my function. It's a really good library. Very simple and convenient to use; But I still have some questions to ask: how to make JSON file of each Chinese character, the data in it looks like the PATH data. I want to try to draw English letters according to this idea.

**Comment by chanind** [comment]
Yes that's exactly right - the JSON data has SVG paths, and then medians which define how the stroke are animated. English letters are hard to do though because a lot of the strokes cross over themselves. the way HanziWriter does animation it won't be able to animate those strokes properly.

**Comment by jackieron** [comment]
> Yes that's exactly right - the JSON data has SVG paths, and then medians which define how the stroke are animated. English letters are hard to do though because a lot of the strokes cross over themselves. the way HanziWriter does animation it won't be able to animate those strokes properly.


Have you ever thought about developing a letter-drawing program, which is very useful. Because many people don't know the correct writing order of letters.
